### PR TITLE
Added checkstyle rules, and made the repo multi modules.

### PR DESCRIPTION
**Pull request by tmatsuo**

### CODE REVIEW

File: appengine-push/pom.xml [hunk 1]
```diff
@@ -56,7 +95,7 @@
       <plugin>
         <groupId>com.google.appengine</groupId>
         <artifactId>gcloud-maven-plugin</artifactId>
-        <version>1.9.17</version>
+        <version>1.9.17a</version>
```
Comment (lesv) [inline appengine-push/pom.xml]: 0.9.58.v20150505

File: cmdline-pull/src/main/java/com/google/cloud/pubsub/client/demos/cli/Main.java [hunk 2]
```diff
@@ -98,7 +102,7 @@ public static void help() {
                         + "PROJ create_topic TOPIC\n"
                         + "PROJ delete_topic TOPIC\n"
                         + "PROJ list_subscriptions\n"
-                        + "PROJ create_subscription SUBSCRIPTION LINKED_TOPIC\n"
+                        + "PROJ create_subscription SUBSCRIPTION LINKED_TOPIC [PUSH_ENDPOINT]\n"
```
Comment (lesv) [inline cmdline-pull/src/main/java/com/google/cloud/pubsub/client/demos/cli/Main.java]: Line length?

File: cmdline-pull/src/main/java/com/google/cloud/pubsub/client/demos/cli/SubscriptionMethods.java [hunk 3]
```diff
@@ -26,6 +30,9 @@ public static void createSubscription(Pubsub client, String[] args)
         Subscription subscription = new Subscription()
                 .setTopic(PubsubUtils.getFullyQualifiedResourceName(
                         PubsubUtils.ResourceType.TOPIC, args[0], args[3]));
+        if (args.length == 5) {
+            subscription = subscription.setPushConfig(new PushConfig().setPushEndpoint(args[4]));
```
Comment (lesv) [inline cmdline-pull/src/main/java/com/google/cloud/pubsub/client/demos/cli/SubscriptionMethods.java]: Line length?

**Comment by lesv** [comment]
PTAL
1. Validate checkstyle rules to Cloud Style Guide and 80 character line length.
2. Add back appengine plugin.

**Comment by tmatsuo** [comment]
@lesv PTAL

**Comment by lesv** [comment]
LGTM

**Pull request merged**